apps/aecore/src/aec_db_gc.erl

```diff
@@ -604,6 +608,42 @@ update_current_scan(Hash, Scanners, #st{current_scan = #{hashes := Hashes} = Sca
 %%                 [aec_blocks:height(MicroBlock), N, Res]),
 %%     Res.
 
+db_safe_access_scan() ->
+  case lists:usort([ element(1, db_safe_access_scan(Tree))
+                     || Tree <- [accounts, calls, contracts, oracles, channels, ns] ]) of
+      [ok] -> ok;
+      _    -> error
+  end.
+
+db_safe_access_scan(Tree) ->
+    {ok, Block} = aec_chain:top_key_block(),
+    Height = aec_blocks:height(Block),
+    {ok, Hash} = aec_blocks:hash_internal_representation(Block),
+    case get_mpt(Tree, Hash) of
+        empty ->
+            ?DBG("Tree ~p empty at height ~p - no scan needed", [Tree, Height]),
+            {ok, 0};
+        MPT ->
+            ?DBG("DB safe access FULL_SCAN for ~p at height ~p from root hash ~w",
+                 [Tree, Height, aeu_mp_trees:root_hash(MPT)]),
+            OldDBSafeAccess = persistent_term:get({aec_db, db_safe_access}, undefined),
+            [ persistent_term:put({aec_db, db_safe_access}, true) || OldDBSafeAccess /= true ],
+            try
+                N = visit_reachable(MPT),
+                ?DBG("DB safe access FULL_SCAN for ~p completed, checked ~p nodes", [Tree, N]),
+                {ok, N}
+            catch E:{Reason, _S} ->
+                ?DBG("DB safe access FULL_SCAN on ~p failed ~p:~p", [Tree, E, Reason]),
+                {error, Reason}
+            after
+                case OldDBSafeAccess of
+                    undefined -> persistent_term:erase({aec_db, db_safe_access});
+                    false     -> persistent_term:put({aec_db, db_safe_access}, false);
```

Comment:
```suggestion
                    true     -> persistent_term:put({aec_db, db_safe_access}, OldDBSafeState)
```
Don't you just want to put back what was there before, unless there was nothing?